posydon/binary_evol/DT/step_detached.py

```diff
@@ -605,328 +676,219 @@ def get_mist0(self, star, htrack):
-                                     posydon_attribute[0])) ** 2
-                            + (sc_center_he4_H.transform(
-                                get_track_val(MESA_label[1], htrack, *x))
-                               - sc_center_he4_H.transform(
-                                     posydon_attribute[1]))
-                            ** 2 + (sc_log_R_H.transform(
-                                get_track_val(MESA_label[2], htrack, *x))
-                                - sc_log_R_H.transform(
-                                     posydon_attribute[2])) ** 2,
-                            x0, method="TNC", bounds=bnds,
-                        )
-                        if ((self.get_track_val("mass", star.htrack, *sol.x)
-                                - self.get_track_val(
-                                    "he_core_mass", star.htrack, *sol.x))
-                                / self.get_track_val(
-                                    "mass", star.htrack, *sol.x) >= 0.05):
-                            initials = (np.nan, np.nan)
-
-            if initials is None:
-                if self.verbose:
-                    print("sol.fun = ", np.abs(sol.fun))
-                if np.abs(sol.fun) < tolerance_mist_integration:
-                    if self.verbose:
-                        print("minimization in matching considered acceptable,"
-                              " with", np.abs(sol.fun), "<",
-                              tolerance_mist_integration, "tolerance")
-                    initials = sol.x
-                    star.fun = sol.fun
-                    star.stiching_rel_mass_difference = (
-                        self.get_track_val("mass", htrack, *sol.x) - star.mass
-                    ) / star.mass
-                    if star.log_R in MESA_label:
-                        star.stiching_rel_logRadius_difference = (
-                            self.get_track_val("log_R", htrack, *sol.x)
-                            - star.log_R) / star.log_R
-                    else:
-                        star.stiching_rel_logRadius_difference = np.nan
-                    if (star.total_moment_of_inertia is not None
-                            and not np.isnan(star.total_moment_of_inertia)):
-                        star.stiching_rel_inertia_difference = (
-                            self.get_track_val("inertia", htrack, *sol.x)
-                            - star.total_moment_of_inertia
-                        ) / star.total_moment_of_inertia
+                    elif star.state in LIST_ACCEPTABLE_STATES_FOR_postMS:
+                        star.htrack = False
+                    htrack  = star.htrack
+                    x0 = get_root0(
+                        MESA_label, posydon_attribute, htrack, rs=rs)
+                    # bnds = ([m_min_H, m_max_H], [0, None])
+                    sol = minimize(square_difference,
+                                   x0, method="TNC", bounds=bnds)
```

Comment:
Since we change the grids, we should update **list_for_matching** for new MESA_label, posydon_attribute,  bnds as well. Just like the code above in 2nd alternative matching.